src/app/hooks/useViewTracker/index.jsx

```diff
@@ -31,12 +28,30 @@ const useViewTracker = (props = {}) => {
     statsDestination,
   } = useContext(EventTrackingContext);
   const { service } = useContext(ServiceContext);
+  const initObserver = async () => {
```

Comment:
I'm not sure about a separate component but I could extract it into a separate function. I'd keep it in the same file though because the logic is specific to this hook and I don't want to expose it to reuse.

Reply:
Actually this will be tricky because the function will need the hook state setter `setIsInView` and the observer instance stored in a ref.

Reply:
Yeah, maybe leave this as it is 👍